scripts/baseline_comparison/baselines.py

```diff
@@ -34,110 +28,99 @@ class ResultRow:
-    # we expect a tensor of results with shape (n_tasks, n_folds)
-    assert metric_errors.shape == (len(folds),)
     rows = []
     for fold in folds:
-        task = repo.task_name(dataset=dataset, fold=fold)
-        metric_error = metric_errors.loc[(dataset, fold)]
-        config_weights = ensemble_weights.loc[(dataset, fold)]
-
-        # select configurations used in the ensemble as infer time only depends on the models with non-zero weight.
-        config_selected_ensemble = [
-            config
-            for config, weight in zip(config_selected, config_weights)
-            if weight != 0
-        ]
-        runtimes = get_runtime(
-            repo=repo,
-            tid=tid,
-            fold=fold,
-            config_names=config_sampled,
-            runtime_col='time_train_s',
-            fail_if_missing=False,
-        )
-        latencies = get_runtime(
-            repo=repo,
-            tid=tid,
+        df_metrics, ensemble_weights = repo.evaluate_ensemble(
+            dataset=dataset,
             fold=fold,
-            config_names=config_selected_ensemble,
-            runtime_col='time_infer_s',
-            fail_if_missing=False,
+            configs=configs,
+            fit_order=fit_order,
+            seed=seed,
+            ensemble_size=ensemble_size,
+            ensemble_kwargs=ensemble_kwargs,
+            time_limit=time_limit,
+            rank=False,
         )
+        assert len(df_metrics) == 1
+        metrics = df_metrics.iloc[0]
+        configs_selected = [c for c in configs if c in ensemble_weights.columns]
+
+        task = repo.task_name(dataset=dataset, fold=fold)
+
+        metric_error = metrics["metric_error"]
+        metric_error_val = metrics["metric_error_val"]
+        time_train_s = metrics["time_train_s"]
+        time_infer_s = metrics["time_infer_s"]
+
         rows.append(ResultRow(
             dataset=dataset,
             fold=fold,
             method=method,
             test_error=metric_error,
             rank=rank_scorer.rank(task, metric_error),
             normalized_error=normalized_scorer.rank(task, metric_error),
-            time_train_s=sum(runtimes.values()),
-            time_infer_s=sum(latencies.values()),
-            config_selected=config_sampled,
+            time_train_s=time_train_s,
+            time_infer_s=time_infer_s,
+            metric_error_val=metric_error_val,
+            config_selected=configs_selected,
             seed=seed,
+            metadata=dict(
+                n_iterations=ensemble_size,
+                time_limit=time_limit,
+            )
         ))
     return rows
 
 
-def framework_name(framework_type, max_runtime=None, ensemble_size=default_ensemble_size, tuned: bool=True) -> str:
+def framework_name(framework_type, max_runtime=None, ensemble_size=default_ensemble_size, tuned: bool=True, all: bool = False, prefix: str = None) -> str:
     method = framework_type if framework_type else "All"
+    if prefix is None:
```

Comment:
Does the scripts to analyse the results from the paper still work with those changes? You mention the results are the same float wise, I wonder if the names are also compatible (If not we should mention this in the readme)

Reply:
yes they do work and are unchanged. prefix and all are always `None` and `False` for evaluate_baselines.py. I have other code that actually sets these to non-default values when I was testing TabPFNMix, but that can be a follow-up PR as it would change `evaluate_baselines.py`.

Reply:
The names are identical